File: 2021/python/2.py
def main():
    x, y = 0, 0
    with open("../inputs/2.txt") as input:
        for line in input:
            direction, magnitude = line.split()
            if direction == "forward":
                x += int(magnitude)
            elif direction == "down":
                y += int(magnitude)
            elif direction == "up":
                y -= int(magnitude)
            else:
                raise ValueError()

    print(x * y)


if __name__ == "__main__":
    main()

File: 2021/julia/2.jl
function move_ship(file)
    pos, depth = 0, 0
    open("../inputs/2.txt", "r") do input
        for line in eachline(input)
            direction, magnitude = split(line)
            mag_int = parse.(Int, magnitude)
            if direction == "forward"
                pos += mag_int
            elseif direction == "down"
                depth += mag_int
            elseif direction == "up"
                depth -= mag_int
            else
                error("Invalid direction")
            end;
        end;
    end;
    return pos * depth
end;

function main()
    println(move_ship("../inputs/2.txt"))
end;

main()